bundles/com.espressif.idf.core/src/com/espressif/idf/core/toolchain/enums/Target.java

```diff
@@ -0,0 +1,110 @@
+/*******************************************************************************
+ * Copyright 2025 Espressif Systems (Shanghai) PTE LTD. All rights reserved.
+ * Use is subject to license terms.
+ *******************************************************************************/
+package com.espressif.idf.core.toolchain.enums;
+
+import java.util.Locale;
+
+/**
+ * Target enum for ESP-IDF toolchains
+ * 
+ * @author Ali Azam Rana
+ *
+ */
+public enum Target
+{
+	ESP32("esp32", Architecture.XTENSA), //$NON-NLS-1$
+	ESP32S2("esp32s2", Architecture.XTENSA), //$NON-NLS-1$
+	ESP32S3("esp32s3", Architecture.XTENSA), //$NON-NLS-1$
+	ESP32C2("esp32c2", Architecture.RISCV32), //$NON-NLS-1$
+	ESP32C3("esp32c3", Architecture.RISCV32), //$NON-NLS-1$
+	ESP32C6("esp32c6", Architecture.RISCV32), //$NON-NLS-1$
+	ESP32H2("esp32h2", Architecture.RISCV32), //$NON-NLS-1$
+	ESP32P4("esp32p4", Architecture.RISCV32); //$NON-NLS-1$
+
+	private static final String CMAKE_TOOLCHAIN_FILE_TEMPLATE = "toolchain-%s.cmake"; //$NON-NLS-1$
+
+	private final String idfName;
+	private final Architecture arch;
+
+	Target(String idfName, Architecture arch)
+	{
+		this.idfName = idfName;
+		this.arch = arch;
+	}
+
+	public String idfName()
+	{
+		return idfName;
+	}
+
+	public Architecture arch()
+	{
+		return arch;
+	}
+
+	public static Target fromString(String s)
+	{
+		if (s == null)
+			return null;
+		String key = s.toLowerCase(Locale.ROOT);
+		for (Target t : values())
```

Comment:
If a value isn’t present in the enum but comes from the IDF preview list, our current approach returns null and skips it. That doesn’t align with the PR’s goal

Reply:
@sigmaaa kindly take a look again the code is updated now